### PR TITLE
Adding a Windows Server 2012 Vagrantfile and Set-Up instructions

### PR DESCRIPTION
**Pull request by Willsparker**
Having a Vagrant file for Windows Server 2012 allows the testing of the Windows playbook on a local VM. The README was updated to contain instructions on how to set this up, and subsequently run the playbook on the VM. The Vagrantfile takes care of configuring the VM to allow winRM connections, so only changes to the host machine is needed.

Fixes: #864

### CODE REVIEW

File: ansible/README.md [hunk 1]
```diff
@@ -97,6 +97,19 @@ Yes, in order to access the package repositories (we will perform either `yum in
 
 4) The Ansible playbook will download and install any dependencies needed to build OpenJDK
 
+## How do I run the playbook on a Windows Machine?
+
+As Ansible can't run on Windows, it has to run remotely. To test the playbook, a Vagrant VM can be used.
```
Comment (karianna) [inline ansible/README.md]: Can run on windows so you have to run remotely?  Doesn't quite make sense to me....

File: ansible/README.md [hunk 2]
```diff
@@ -97,6 +97,19 @@ Yes, in order to access the package repositories (we will perform either `yum in
...
+You can do this by following these steps:
+
+1) `ln -sf Vagrantfile.WindowsS12 Vagrantfile && vagrant up` in the `../ansible` directory
+2) Note the IP address that is output from running `vagrant up` and place it into a text file. e.g. `hosts.win`. Place this text file in `../anisble/playbooks/AdoptopenJDK_Windows_Playbook`
```
Comment (karianna) [inline ansible/README.md]: type in the directory path for ansible

File: ansible/Vagrantfile.WindowsS12 [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# vi: set ft=ruby :
+
+$script = <<SCRIPT
+Start-Process powershell -Verb runAs
+wget https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1 -OutFile .\\ConfigureRemotingForAnsible.ps1
+.\\ConfigureRemotingForAnsible.ps1 -EnableCredSSP
+ipConfig
+SCRIPT
+
+Vagrant.configure("2") do |config|
```
Comment (karianna) [inline ansible/Vagrantfile.WindowsS12]: 2 is a magic number

File: ansible/Vagrantfile.WindowsS12 [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+$script = <<SCRIPT
```
Comment (karianna) [inline ansible/Vagrantfile.WindowsS12]: Comment at the top of this block explaining roughly what it does

File: ansible/README.md [hunk 5]
```diff
@@ -97,6 +97,20 @@ Yes, in order to access the package repositories (we will perform either `yum in
 
 4) The Ansible playbook will download and install any dependencies needed to build OpenJDK
 
+## How do I run the playbook on a Windows Machine?
+
+As Ansible can't run on Windows, it has to be run on a seperate system and then pointed at a Windows machine (such as a VM). To test the playbook, a vagrant VM can be used.
```
Comment (karianna) [inline ansible/README.md]: Thanks for the update, but I'll still bite here :-) - https://www.ansible.com/integrations/infrastructure/windows why can't we run on windows directly?

Comment (Willsparker) [reply ansible/README.md]: I was going off what the FAQs say in the Ansible documentation:
https://docs.ansible.com/ansible/latest/user_guide/windows_faq.html

Also doing it this way seemed easier than dealing with the problems found in issue #791

Comment (sxa555) [reply ansible/README.md]: @karianna That page is talking about ansible's Windows modules that you can use when remotely connecting in, not executing the playbooks natively on the host. I believe it should be doable but it's a bit of a pig to set up. We have https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/791 to cover anything we can do to make this easier in the future, but at the moment we don't easily support it or test it.

File: ansible/Vagrantfile.Win2012 [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+Start-Process powershell -Verb runAs
+wget https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1 -OutFile .\\ConfigureRemotingForAnsible.ps1
+.\\ConfigureRemotingForAnsible.ps1 -EnableCredSSP
+ipConfig
+SCRIPT
+
+# 2 = version of configuration file for Vagrant 1.1+ leading up to 2.0.x
+Vagrant.configure("2") do |config|
+
+  config.vm.define :adoptopenjdkW10 do |adoptopenjdkW10|
```
Comment (sxa555) [inline ansible/Vagrantfile.Win2012]: Should probably be `W2012` since this isn't Windows 10

File: ansible/README.md (hunk 5 above)
Comment (karianna) [reply ansible/README.md]: Got it thanks both.

**Pull request merged**